eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -285,17 +285,17 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
           context.system.eventStream.publish(ShortChannelIdAssigned(self, normal.channelId, normal.channelUpdate.shortChannelId, None))
 
           // we rebuild a new channel_update with values from the configuration because they may have changed while eclair was down
-          // NB: we don't update the routing fees, because we don't want to overwrite manual changes made with CMD_UPDATE_RELAY_FEE
-          // Since CMD_UPDATE_RELAY_FEE is handled even when being offline, that's the preferred solution to update routing fees
+          val defaultFees = nodeParams.relayParams.defaultFees(data.commitments.announceChannel)
+          val (feeBase, feeProportionalMillionth) = nodeParams.db.relayFees.getFees(remoteNodeId).getOrElse((defaultFees.feeBase, defaultFees.feeProportionalMillionth.toLong))
```

Comment:
This should be factored in `NodeParams` exactly like per-node features and we should use the new `RelayFees` class:

```scala
class NodeParams(...) {
  def featuresFor(nodeId: PublicKey): Features = overrideFeatures.getOrElse(nodeId, features)
  def relayFeesFor(nodeId: PublicKey, announceChannel: Boolean): RelayFees = ???
}
```

Re: the change in behavior (deleted comment), I understand why you went that way, but it means we will have to be careful and always edit `eclair.conf` after calling `./eclair-cli updaterelayfee`.

Another way, closer to the origin behavior, would be to use the hardcoded fee if there are no existing channel, and otherwise keep the existing fee. I think it would be easier from an operational perspective. That way, the order of precedence would be clear (api > eclair.conf), whereas in your proposal it depends on when eclair was last restarted. WDYT?

Reply:
> Another way, closer to the origin behavior, would be to use the hardcoded fee if there are no existing channel, and otherwise keep the existing fee. I think it would be easier from an operational perspective. That way, the order of precedence would be clear (api > eclair.conf), whereas in your proposal it depends on when eclair was last restarted. WDYT?

Thinking about this, I see the complication with getting the feerate used by other channels to the same peer. This would have to be handled in the `Peer` class. OTOH, it would make total sense given the intent to make relay fees peer-centric and not channel-centric.

Reply:
There is no need to edit `eclair.conf` all the time. The API always gets precedence over `eclair.conf`.

- In case we manually change fees using the API, then it will be recorded in `RelayFeesDb` and it will survive restarts.
- In case we never changed the fees manually, then it will use the default fees from `eclair.conf`. This means that if we edit `eclair.conf`, all channels with the default fees will be updated without the need for manual intervention.

The rationale is that for most nodes we will use the default fees and only for a few important nodes we will manually change it.

Reply:
I see now. I thought the per-node setting was in eclair.conf but it was actually retrieved from the db, making the behavior the same as now.